source/postprocess/boundary_velocity_residual_statistics.cc

```diff
@@ -0,0 +1,419 @@
+/*
+  Copyright (C) 2011 - 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+
+#include <aspect/postprocess/boundary_velocity_residual_statistics.h>
+#include <aspect/utilities.h>
+#include <aspect/geometry_model/interface.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    BoundaryVelocityResidualStatistics<dim>::execute (TableHandler &statistics)
+    {
+      // create a quadrature formula for the velocity.
+      const QGauss<dim-1> quadrature_formula (this->introspection().polynomial_degree.velocities+1);
+
+      FEFaceValues<dim> fe_face_values (this->get_mapping(),
+                                        this->get_fe(),
+                                        quadrature_formula,
+                                        update_values |
+                                        update_JxW_values |
+                                        update_quadrature_points);
+
+      std::vector<Tensor<1,dim> > velocities (fe_face_values.n_quadrature_points);
+
+      std::map<types::boundary_id, double> local_max_vel;
+      std::map<types::boundary_id, double> local_min_vel;
+      std::map<types::boundary_id, double> local_velocity_square_integral;
+      std::map<types::boundary_id, double> local_boundary_area;
+
+      std::set<types::boundary_id> boundary_indicators;
+      boundary_indicators.insert(this->get_geometry_model().translate_symbolic_boundary_name_to_id("top"));
+
+      for (const auto p : boundary_indicators)
+        {
+          local_max_vel[p] = -std::numeric_limits<double>::max();
+          local_min_vel[p] = std::numeric_limits<double>::max();
+        }
+
+      // for every surface face that is part of a geometry boundary
```

Comment:
I would update that this only happens for every *selected* geometry boundary (which happens to be only "top" at the moment).